### PR TITLE
backport: trivial 2024 10 25 pr3

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Trivial backports

## How Has This Been Tested?
built locally

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/net.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ static const bool DEFAULT_FIXEDSEEDS = true;
 static const size_t DEFAULT_MAXRECEIVEBUFFER = 5 * 1000;
 static const size_t DEFAULT_MAXSENDBUFFER    = 1 * 1000;
 
-static constexpr bool DEFAULT_V2_TRANSPORT{false};
+static constexpr bool DEFAULT_V2_TRANSPORT{true};
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/net.h:117]: Hmmm, I think I wanna drop this anyhow. I don't think we want v22 final having default v2. only maybe a 22.1?

**Comment by PastaPastaPasta** [comment]
dropped non-trivial commits

**Review by UdjinM6 — DISMISSED** [review]
utACK 66566f30d2363cf55dcf272f0776bc00d103e539 but also needs confirmation that Guix is ok (GH had and issue, couldn't reproduce it on my machine though)

**Review by kwvg — APPROVED** [review]
utACK 66566f30d2363cf55dcf272f0776bc00d103e539

**Comment by UdjinM6** [comment]
25677 need a fix:
```diff
diff --git a/src/node/interfaces.cpp b/src/node/interfaces.cpp
index e1418aa442..f15026cf7b 100644
--- a/src/node/interfaces.cpp
+++ b/src/node/interfaces.cpp
@@ -767,7 +767,7 @@ public:
     bool checkFinalTx(const CTransaction& tx) override
     {
         LOCK(cs_main);
-        return CheckFinalTxAtTip(m_node.chainman->ActiveChain().Tip(), tx);
+        return CheckFinalTxAtTip(*Assert(m_node.chainman->ActiveChain().Tip()), tx);
     }
     bool isInstantSendLockedTx(const uint256& hash) override
     {
```

File: test/functional/feature_dbcrash.py [hunk 2]
```diff
@@ -30,17 +30,17 @@
...
-    CTxOut,
 )
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
-    create_confirmed_utxos,
+)
+from test_framework.wallet import (
+    MiniWallet,
+    getnewdestination,
```
Comment (UdjinM6) [inline test/functional/feature_dbcrash.py:43]: 25087: linter complains
```
test/functional/feature_dbcrash.py:41: error: Module 'test_framework.wallet' has no attribute 'getnewdestination' 
```

**Review by UdjinM6 — APPROVED** [review]
utACK fbc0bce2e62592817676408daaeb33384b76aa78

**Pull request merged**